packages/number-field/src/vaadin-number-field.js

```diff
@@ -467,9 +492,14 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
   checkValidity() {
     if (
       this.inputElement &&
-      (this.required || this.min !== undefined || this.max !== undefined || this.__validateByStep)
+      (this.required || this.min !== undefined || this.max !== undefined || this.step !== undefined)
     ) {
-      return this.inputElement.checkValidity();
+      if (this.value == null || this.value == '') {
```

Comment:
The input value should be sanitized so that the field wouldn't get a non-numeric `value` and the validation works as expected:

With the following
`<vaadin-number-field step="0.5"></vaadin-number-field>`

| typed value   |      field value      |  invalid |
|----------|:-------------:|------:|
| "0.5" | "0.5" | false |
| "0,5" | "0,5" | true |
| ".5" | ".5" | false |
| ",5" | "" | false |
| "1-" | "1-" | true |

Reply:
It looks like there are some assumptions about localization in this table. For me `0,5` is certainly a valid value. Same for `,5`, which results in an input value of `0.5`, when comparing the behaviour with a native number input in Firefox.

Reply:
The table represent how the current revision of the PR works

Reply:
Thanks, I will update the validation logic to cover values with `,` separator.